linker/Linker.Steps/MarkStep.cs

```diff
@@ -46,6 +46,7 @@ public partial class MarkStep : IStep {
 		protected Queue<AttributeProviderPair> _assemblyLevelAttributes;
 		protected Queue<AttributeProviderPair> _lateMarkedAttributes;
 		protected List<TypeDefinition> _typesWithInterfaces;
+		protected List<ActivatorCreateInstanceMarkingInformation> _activatorCreateInstanceTypes;
```

Comment:
Why is it not HashSet?

Reply:
Because two `Activator.CreateInstance` usages related to the same type could cause different marking behavior.  The first usage may only mark the `.ctor()` while a later usage may require all `.ctor` be marked.

See the test `CreateAndThenCastAfterAnyCtorSameType` https://github.com/mono/linker/pull/419/files#diff-a25cda93d2795e5473fb811034eda4f4

Another similar scenario is that in our UnityLinker we can detect all stack variables types of a `Activator.CreateInstance` usage and are able to register a `ActivatorCreateInstanceMarkingInformation.ConstructorCollector ` method that will precisely mark just the ctors that are needed.  This is a hypothetical example, but we could do it if a worth while scenario arose and if I were to do it I'd rather not have to debug and figure out why this code failed me.

I could have implemented custom Equals, GetHashCode, ==, etc on `ActivatorCreateInstanceMarkingInformation` and then used a HashSet, but given the following I decided that wasn't worth the effort.

* `ActivatorCreateInstanceMarkingInformation` has a number of properties to consider in determining if two `Activator.CreateInstance` usages should be considered equal.

* Maintaining the overrides for comparing can be error prone.

* I'm assuming the number of detectable `Activator.CreateInstance` usages that we add to this list will be relatively low.  Maybe that's not a fair assumption?